code/game/objects/items/weapons/batons.dm

```diff
@@ -39,27 +46,68 @@
 			user.take_organ_damage(force * 2)
 		return
 
-	if(user.a_intent == INTENT_HARM || isrobot(target)) // Lethal attack or it's a borg (can't knock them down!)
+	if(user.a_intent == INTENT_HARM)
+		return ..()
+	if(on_cooldown)
+		return
+	if(issilicon(target) && !affect_silicon)
 		return ..()
-	else if(!on_cooldown) // Non-lethal attack - knock them down
+	else
+		stun(target, user)
+
+/**
+  * Called when a target is about to be hit non-lethally.
+  *
+  * Arguments:
+  * * target - The mob about to be hit
+  * * user - The attacking user
+  */
+/obj/item/melee/classic_baton/proc/stun(mob/living/target, mob/living/user)
+	if(issilicon(target))
+		user.visible_message("<span class='danger'>[user] pulses [target]'s sensors with [src]!</span>",\
+							 "<span class='danger'>You pulse [target]'s sensors with [src]!</span>")
+		on_silicon_stun(target, user)
+	else
 		// Check for shield/countering
 		if(ishuman(target))
 			var/mob/living/carbon/human/H = target
 			if(H.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK))
-				return
+				return FALSE
 			if(check_martial_counter(H, user))
-				return
-		// Visuals and sound
-		user.do_attack_animation(target)
-		playsound(target, 'sound/effects/woodhit.ogg', 75, TRUE, -1)
-		add_attack_logs(user, target, "Stunned with [src]")
-		target.visible_message("<span class='danger'>[user] has knocked down [target] with \the [src]!</span>", \
-							   "<span class='userdanger'>[user] has knocked down [target] with \the [src]!</span>")
-		// Hit 'em
-		target.LAssailant = iscarbon(user) ? user : null
-		target.Weaken(3)
-		on_cooldown = TRUE
-		addtimer(CALLBACK(src, .proc/cooldown_finished), BATON_STUN_COOLDOWN)
+				return FALSE
+		user.visible_message("<span class='danger'>[user] knocks down [target] with [src]!</span>",\
+							 "<span class='danger'>You knock down [target] with [src]!</span>")
+		on_non_silicon_stun(target, user)
+	// Visuals and sound
+	user.do_attack_animation(target)
+	playsound(target, stun_sound, 75, TRUE, -1)
+	add_attack_logs(user, target, "Stunned with [src]")
+	// Hit 'em
+	target.LAssailant = iscarbon(user) ? user : null
+	target.Weaken(stun_time)
+	on_cooldown = TRUE
+	addtimer(CALLBACK(src, .proc/cooldown_finished), cooldown)
+	return TRUE
+
+/**
+  * Called when a silicon has been stunned.
+  *
+  * Arguments:
+  * * target - The hit mob
+  * * user - The attacking user
+  */
+/obj/item/melee/classic_baton/proc/on_silicon_stun(mob/living/silicon/target, mob/living/user)
+	target.flash_eyes(affect_silicon = TRUE)
+
+/**
+  * Called when a non-silicon has been stunned.
+  *
+  * Arguments:
+  * * target - The hit mob
+  * * user - The attacking user
+  */
+/obj/item/melee/classic_baton/proc/on_non_silicon_stun(mob/living/target, mob/living/user)
+	return
```

Comment:
Not personally, but I think it'd be good to have because it means less code to write/alter if someone wants to add a new baton with specific functionality.